tt_metal/hw/ckernels/grayskull/metal/llk_api/llk_sfpu/metal_ckernel_sfpu.h

```diff
@@ -456,6 +456,25 @@ inline void calculate_silu()
 
 }
 
+template <bool APPROXIMATION_MODE, int ITERATIONS>
+inline void calculate_tiled_prod()
+{
+    do {
```

Comment:
What's the purpose of this do while loop?

Reply:
The plan is to store the result in a temporary variable `result`.  So I have to declare this outside the for loop as it iterates through all the rows of the dst_reg. But I was not abe to create a vFloat variable outside a looping statement. So Muthu suggested to use a do while loop. But Now I guess the code works without the loop. Will run a few tests and update the code accordingly

Reply:
I have updated the code for GS. The sae should work for WH_B0, but still would test is there and update the code once I get the machine

Reply:
Updated